### PR TITLE
Proposal

### PR DESCRIPTION
**Pull request by GMagician**
This is a proposal of a code uniforming, please let me know

- ~~Remove mixing 2/4 tab spaces (default to 2)~~
- use pragma once in include files
- try to group up differences between MCU (CommonMacros.h)

### CODE REVIEW

**Comment by technik-gegg** [comment]
Can you please rework your proposal without the "tabs/spaces/formatting" thingy?
I'm absolutely serious, I'm not going to change this. If someone get's bothered by the formatting, there's a shortcut in VSCode (and most other editors) Shift-Alt-F or "Format Document" which does that for you.

**Comment by GMagician** [comment]
Here same code with no "tabs" changes

**Comment by GMagician** [comment]
Another approach could be to split "timer" implementation into different files, one for each MPU (something done in Marlin with HALs) and have a singleton class to be called in the code

File: src/ClickEncoder.cpp [hunk 1]
```diff
@@ -176,8 +168,8 @@ void ClickEncoder::service(void)
     }
 
     if (doubleClickTicks > 0) {
-      doubleClickTicks--;
```
Comment (GMagician) [inline src/ClickEncoder.cpp:179]: Old and new code is buggy when doubleClickTicks is odd. Also variable is unsigned so never < 0. A different test must be done I think

**Review by technik-gegg — APPROVED** [review]

File: src/ClickEncoder.cpp (hunk 1 above)
Comment (GMagician) [reply src/ClickEncoder.cpp:179]: @technik-gegg what is correct code here? I think it should be just one dec? Right?

Comment (technik-gegg) [reply src/ClickEncoder.cpp:179]: I'd say the there's only one decrement needed, since all this code is meant to trigger for the 600ms, after which a **single click** is generated if there was no 2nd click in the meantime. Not sure why the author decided to add a second decrement here, which effectively reduces the trigger time to 300ms.

**Comment by GMagician** [comment]
Some code change lead to save some memory but I'm wondering: AVR and ESP32 don't compile and former isn't supported anymore. Have a sense to keep all those  #ifdef for such platform?

RAM:   [========  ]  82.7% (used 40672 bytes from 49152 bytes)
Flash: [=======   ]  66.7% (used 174788 bytes from 262144 bytes)

RAM:   [========  ]  82.7% (used 40664 bytes from 49152 bytes)
Flash: [=======   ]  66.7% (used 174748 bytes from 262144 bytes)

**Comment by technik-gegg** [comment]
> Some code change lead to save some memory but I'm wondering: AVR and ESP32 don't compile and former isn't supported anymore. Have a sense to keep all those #ifdef for such platform?

You're right, I'm already planing to kick those out in version 3 of the firmware, so it'll become STM32 only.

**Comment by GMagician** [comment]
May I proceed here with such "cleanup"?

**Comment by technik-gegg** [comment]
Please don't just yet. I'm still working on cleaning out some bugs for 2.13.

**Comment by GMagician** [comment]
Ok I'll stop here. You may merge these into your "2.13" code.
- pragma once in every .h (removed define used only to gett same behaviour)
- removed some unused vars in some c++ code (free some memory)
- some variables inside 'define' to free memory when not used
- some test change to prevent negative indexes to write outside array

File: src/Menus.cpp [hunk 2]
```diff
@@ -1320,18 +1314,12 @@ void showDisplayMenu(char* menuTitle) {
...
-      #if defined(MULTISERVO)
-      if(writeServoMapping()) {
-        stat = true;
-      }
-      #else
-        stat = true;
-      #endif
-    }
-  }
+  bool stat = writeConfig()
```
Comment (technik-gegg) [inline src/Menus.cpp]: Please undo this change.
This is on purpose. If writeConfig returns  false, this means something is wrong with the SD-Card or memory and it doesn't make sense moving on.

File: src/Menus.cpp [hunk 3]
```diff
@@ -1374,7 +1362,6 @@ void showSettingsMenu(char* menuTitle) {
     stopMenu = checkStopMenu(startTime);
 
     current_selection = display.userInterfaceSelectionList(menuTitle, current_selection, _menu);
-    uint8_t fnc = menuOrdinals[current_selection];
```
Comment (technik-gegg) [inline src/Menus.cpp]: Although you're saving one byte of stack memory, I might need the function code for further operations in upcoming versions.

File: src/SMuFF.cpp [hunk 4]
```diff
@@ -385,7 +384,7 @@ void setup() {
   setupHBridge();
   getStoredData();                    // read EEPROM.DAT from SD-Card; this call must happen after setupSteppers()
   //__debug(PSTR("readSequences start"));
-  uint16_t now = millis();
```
Comment (technik-gegg) [inline src/SMuFF.cpp:388]: On STM32 millis() returns a uint32_t. At least that's what the documentation says.

File: src/ZFan.cpp [hunk 5]
```diff
@@ -34,29 +34,28 @@ void ZFan::detach() {
...
 void ZFan::setIndex(int8_t fanIndex) {
-  if(fanIndex != -1 && fanIndex < MAX_FANS) {
+  if(fanIndex >= 0 && fanIndex < MAX_FANS) {
     _fanIndex = fanIndex;
     fanInstances[_fanIndex] = this;
   }
 }
 
 void ZFan::setFanSpeed(uint8_t speed) {
-  _speed = speed;
```
Comment (technik-gegg) [inline src/ZFan.cpp]: _speed is an internal member storing the current speed. Although it's not implemented yet, it's meant to retrieve the current speed setting using getFanSpeed().

File: src/ZServo.cpp [hunk 6]
```diff
@@ -186,8 +181,7 @@ void isrServoTimerHandler() {
   // call all handlers for all servos periodically if the
   // internal timer is being used.
   for(int8_t i=0;  i< MAX_SERVOS; i++) {
-    if(servoInstances[i] == nullptr || servoInstances[i]->isTimerStopped())
```
Comment (technik-gegg) [inline src/ZServo.cpp]: What's the benefit of this change?

File: src/ZStepperLib.cpp [hunk 7]
```diff
@@ -38,11 +38,11 @@ ZStepper::ZStepper(int8_t number, char* descriptor, int8_t stepPin, int8_t dirPi
   _acceleration = acceleration;
   _minStepInterval = minStepInterval;
   if(_stepPin != -1)
-    pinMode(_stepPin,    OUTPUT);
```
Comment (technik-gegg) [inline src/ZStepperLib.cpp]: What did I tell you about tabs and spaces?

**Review by technik-gegg — CHANGES_REQUESTED** [review]

File: src/SMuFF.cpp (hunk 4 above)
Comment (GMagician) [reply src/SMuFF.cpp:388]: It was uint16_t and now it's uint32_t
